federationtypes.go

```diff
@@ -347,9 +351,18 @@ func (r *RespSendJoin) UnmarshalJSON(data []byte) error {
 	return nil
 }
 
+// ToRespState return new RespState with the same data from the given RespSendJoin
```

Comment:
```suggestion
// ToRespState returns a new RespState with the same data from the given RespSendJoin
```

Reply:
I will make a new commit soon.

Reply:
You can also just use GitHub's accept suggestion feature from web.

Reply:
Great! 😃 I wasn't aware of this feature . I made a few additional changes (comments and tests)